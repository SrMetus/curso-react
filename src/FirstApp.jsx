// const newMessage = [1,2,3,4,5,6,7,8,9]
// const newMessage2 = {
//     message: 'Hola mundos',
//     title: 'Sebastian',
// };

// const date = new Date();
// const [month, day, year, hour, minute, second] = [
//     date.getMonth() + 1, 
//     date.getDate(),
//     date.getFullYear(),
//     date.getHours(),
//     date.getMinutes(),
//     date.getSeconds()
// ];

import PropTypes from 'prop-types';

export const FirstApp = ({title, subTitle, name}) => {

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            {/* <h2>{`${day}-${month}-${year} at ${hour}:${minute}:${second}`}</h2>
            <h1>{newMessage}</h1>
            <code>{JSON.stringify(newMessage2)}</code> 
            <p>Soy un subtitulo</p>*/}
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
};

