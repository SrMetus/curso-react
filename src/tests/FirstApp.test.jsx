import { getAllByAltText, getByTestId, render } from '@testing-library/react';
import { FirstApp } from '../FirstApp';


describe('Pruebas en <FirstApp />', () => { 
    test('debe hacer match con el snapshot', () => {
        const title = 'Hola, soy Vegeta';
        const {container} = render(<FirstApp title={title} />);

        expect(container).toMatchSnapshot();

    });
    test('debe mostrar el titulo en un h1', () => {
        const title = 'Hola, soy Vegeta';
        const {container, getByText, getByTestId} = render(<FirstApp title={title} />);
        
        expect(getByText(title)).toBeTruthy();
        // const h1 = container.querySelector('h1');
        // expect(h1.innerHTML).toContain(title);
        expect(getByTestId('test-title').innerHTML).toBe(title);

    });
    test('debe mostrar un subtitulo enviado por props', () => {
        const title = 'Hola, soy Vegeta';
        const subTitle = 'soy un subtitulo'
        const {getByText, getAllByText} = render(
        <FirstApp 
            title={title}
            subTitle={subTitle}
        />);
        // expect(getByText(subTitle)).toBeTruthy();
        expect(getAllByText(subTitle).length).toBe(2);
    })

 });