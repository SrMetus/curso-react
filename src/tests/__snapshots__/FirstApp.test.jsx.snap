// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy Vegeta
  </div>
  <p />
  <p />
  <p />
</div>
`;
